Makefile

```diff
@@ -545,8 +545,11 @@ endif
 # Keep open the possibility of shared libraries in future with, e.g., .so suffix
 #
 ifneq ($(wildcard $(PIO_LIB)/libpio\.*), )
-	LIBS += -lpio
+	ifneq "$(wildcard $(PIO_LIB)/libpio\.*)" "$(PIO_LIB)/libpio.settings"
```

Comment:
My understanding here is:
case 1: only $(PIO_LIB)/libpio.settings exists, so they are equal, and -lpio is not added.
case 2: only $(PIO_LIB)/libpio.a exists, they are not equal, and -lpio is added.
case 3: both files exist, so two items are not equal, so -lpio is added.

why don't we just check for the existence of libpio.a? That seems much more straightforward:
```
ifneq ("$(wildcard $(PIO_LIB)/libpio\.a)","")
```
I got the format from http://www.humbug.in/2012/makefile-check-if-a-file-exists-using-wildcard-function/

Would it ever be named something other than `libpio.a`? I think that could replace the first ifneq on line 547 as well.

Reply:
I see. We are leaving it open to libpio.a or libpio.so. So we could do this:
```
 544 # Depending on PIO version, libraries may be libpio.a, or libpiof.a and libpioc.a
 545 # Keep open the possibility of shared libraries in future with, e.g., .so suffix
 546 #
 547 ifneq ($(wildcard $(PIO_LIB)/libpio\.a), )
 548     LIBS += -lpio
 549 else ifneq ($(wildcard $(PIO_LIB)/libpio\.so), )
 550     LIBS += -lpio
 551 endif
```
That is more readable, at least. It looks like make files don't support booleans, which is why I used two ifneq lines. You could read it as "if libpio.a exists" and "if libpio.so exists".

Reply:
> why don't we just check for the existence of libpio.a?

It could be called `libpio.a`, `libpio.so`, or `libpio.dylib`, depending on the system and whether these are static or dynamic libraries.  I don't think we want to lock ourselves into static libraries.

Your second suggestion only covers linux, not OSX, which we also want to support.  I think it's just easier to treat `libpio.settings` as a special case.  I'm open to another suggestion but the nested if was the way I had seen suggested for "and" in a makefile.

Reply:
I see, I understand now. It is just Makefile clumsiness because of three possible suffixes and lack of boolean operators. Yes, considering that, your formulation is fine.